### PR TITLE
feat: introduce generic SecurityToken to handle ApiKey and clientId tokens

### PR DESCRIPTION
**Pull request by marcambier**
**Issue**

https://github.com/gravitee-io/issues/issues/7991

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `1.39.0-feat-jupiterPlans-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/gateway/gravitee-gateway-api/1.39.0-feat-jupiterPlans-SNAPSHOT/gravitee-gateway-api-1.39.0-feat-jupiterPlans-SNAPSHOT.zip)

### CODE REVIEW

**Review by jhaeyaert — CHANGES_REQUESTED** [review]
Some comment about naming. I let you consider them or leave with it if not pertinent.

File: src/main/java/io/gravitee/gateway/jupiter/api/policy/SecurityToken.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import static io.gravitee.gateway.jupiter.api.policy.SecurityToken.TokenType.*;
+
+import io.vertx.core.cli.CLI;
+
+public class SecurityToken {
+
+    public enum TokenType {
+        CLIENT_ID,
+        API_KEY,
+        EMPTY,
```
[the remark this answers is not in the document]
Comment (marcambier) [reply src/main/java/io/gravitee/gateway/jupiter/api/policy/SecurityToken.java]: Yes, I used NONE :+1:

File: src/main/java/io/gravitee/gateway/jupiter/api/policy/SecurityToken.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    private TokenType tokenType;
+
+    private String tokenValue;
+
+    private SecurityToken(TokenType tokenType, String tokenValue) {
```
[the remark this answers is not in the document]
Comment (marcambier) [reply src/main/java/io/gravitee/gateway/jupiter/api/policy/SecurityToken.java]: Nice catch, I turned it into a string

**Pull request merged**